amulet/unauth_model_ownership/defenses/watermark.py

```diff
@@ -147,154 +154,41 @@ def watermark(self):
                     print(
                         f"Train Epoch: {epoch} Loss: {loss.item():.6f} Acc: {acc/total*100:.2f}"
                     )
-                    self.wmtest()
-                    model.train()
+                    self.verify(self.target_model)
+                    self.target_model.train()
         print("Finished Watermark Embeddding")
-        return model
+        return self.target_model
 
-    def wmtest(self):
-        self.model.eval()
+    def verify(self, model: nn.Module, threshold: int = 90):
```

Comment:
return type.

Also, it might be better to have a float threshold, and assume that in this specific case, the accuracy is `0.0-1.0`, to cover things like non-accuracy-based verification methods, e.g. some reconstruction distance for generative models, or p-value based.